### PR TITLE
Update OWLImporter

### PR DESCRIPTION
**Pull request by nikokaoja**
Originally was suppose to be raw rules importer, but then re-used this branch to update OwlImporter and added a few extra validations

### CODE REVIEW

File: tests/tests_unit/rules/importer/test_ontology2rules.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+
+def test_owl2transformation_rules() -> None:
+    # Arrange
+    owl_importer = OWLImporter(wind_energy_ontology)
+    rules = owl_importer.to_rules(make_compliant=True)
+
+    # Assert
+    assert str(rules.metadata.namespace) == "https://kg.cognite.ai/wind/"
+    assert len(rules.classes.data) == 71
```
Comment (doctrino) [inline tests/tests_unit/rules/importer/test_ontology2rules.py:12]: The rules.classes is a user list, this is to access the underlying list. However, you can also just do
```suggestion
    assert len(rules.classes) == 71

```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 2]
```diff
@@ -216,3 +210,24 @@ class InformationRules(RuleModel):
     metadata: InformationMetadata = Field(alias="Metadata")
     properties: SheetList[InformationProperty] = Field(alias="Properties")
     classes: SheetList[InformationClass] = Field(alias="Classes")
+
+    @model_validator(mode="after")
+    def update_entities_prefix(self) -> Self:
+        prefix = self.metadata.prefix
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:216]: Maybe more readable?
```suggestion
        default_prefix = self.metadata.prefix
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 3]
```diff
@@ -216,3 +210,24 @@ class InformationRules(RuleModel):
...
+
+        # update expected_value_types
+        for i in range(0, len(self.properties.data)):
+            if self.properties.data[i].value_type.prefix == "undefined":
+                self.properties.data[i].value_type.prefix = prefix
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:221]:
```suggestion
        for property_ in self.properties:
            if property_.value_type.prefix == "undefined":
                property_.value_type.prefix = prefix
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 4]
```diff
@@ -216,3 +210,24 @@ class InformationRules(RuleModel):
...
+
+        # update parent classes
+        for i in range(0, len(self.classes.data)):
+            if self.classes.data[i].parent:
+                for parent in cast(list[ParentClass], self.classes.data[i].parent):
+                    if parent.prefix == "undefined":
+                        parent.prefix = prefix
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:228]:
```suggestion
        for class_ in self.classes:
            if class_.parent:
                for parent in cast(list[ParentClass], class_.parent):
                    if parent.prefix == "undefined":
                        parent.prefix = prefix
```

File: cognite/neat/rules/_importer/_base.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class BaseImporter(ABC):
+    """
+    BaseImporter class which all importers inherit from.
+    """
+
+    @abstractmethod
+    def __init__(self):
+        ...
```
Comment (doctrino) [inline cognite/neat/rules/_importer/_base.py:15]: Serves no purpose. 
```suggestion
```

File: cognite/neat/rules/_importer/_base.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    @abstractmethod
+    def to_rules(self, *args, **kwargs) -> DomainRules | InformationRules:
```
Comment (doctrino) [inline cognite/neat/rules/_importer/_base.py:18]: Will we ever have an importer that produces domain rules? That seems like a very limited format for rules.

It it possible have more structure here? Can we avoid `args` and `kwargs`?

**Review by doctrino — APPROVED** [review]
I think there is probably some refactoring that can be done to the OWL implementation, however, I assume that is mostly copy-paste from the last version, so I would add a technical debt task for that and continue towards finishing the guide.

File: cognite/neat/rules/_importer/_base.py (hunk 6 above)
Comment (nikokaoja) [reply cognite/neat/rules/_importer/_base.py:18]: Good point, I doubt. They will probably either create information or cdf architect rules

File: cognite/neat/rules/_importer/_base.py (hunk 5 above)
Comment (nikokaoja) [reply cognite/neat/rules/_importer/_base.py]: Removed

**Pull request merged**